src/Compilers/CSharp/Test/Syntax/Parsing/LambdaReturnTypeParsingTests.cs

```diff
@@ -5068,6 +5077,229 @@ public void Dynamic_02()
+                                N(SyntaxKind.DefaultLiteralExpression);
+                                {
+                                    N(SyntaxKind.DefaultKeyword);
+                                }
+                            }
+                        }
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                }
+                EOF();
+            }
+        }
+
+        [Fact]
+        public void Var_04()
+        {
+            string source = "F(var x => x)";
+            verify(source, TestOptions.Regular9);
+            verify(source);
+
+            void verify(string source, ParseOptions? parseOptions = null)
+            {
+                UsingExpression(source, parseOptions,
+                    // (1,7): error CS1003: Syntax error, ',' expected
+                    // F(var x => x)
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "x").WithArguments(",", "").WithLocation(1, 7));
+
+                N(SyntaxKind.InvocationExpression);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "F");
+                    }
+                    N(SyntaxKind.ArgumentList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.Argument);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "var");
+                            }
+                        }
+                        M(SyntaxKind.CommaToken);
+                        N(SyntaxKind.Argument);
+                        {
+                            N(SyntaxKind.SimpleLambdaExpression);
+                            {
+                                N(SyntaxKind.Parameter);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "x");
+                                }
+                                N(SyntaxKind.EqualsGreaterThanToken);
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "x");
+                                }
+                            }
+                        }
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                }
+                EOF();
+            }
+        }
+
+        [Fact]
+        public void Var_05()
+        {
+            string source = "var d = var () => default;";
+            verify(source, TestOptions.Regular9);
+            verify(source);
+
+            void verify(string source, ParseOptions? parseOptions = null)
+            {
+                UsingDeclaration(source, parseOptions,
+                    // (1,9): error CS1031: Type expected
```

Comment:
>error CS1031: Type expected [](http://example.com/codeflow?start=30&length=27)

It feels like this generic error might confuse a user.

Reply:
Perhaps `error CS0825: The contextual keyword 'var' may only appear within a local variable declaration or in script code` is suitable. [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6SA3CAJxIA8BuOh4sgFhIBZABQBKGj3oBfAtKA===)